### ISSUE TITLE
Provide a hook to run custom scripts on startup

### ISSUE THREAD

**Issue by Mahoney** [docker/for-mac#5421]
This is a feature request not a bug report.

I've written some scripts that call `docker events` and react to those events. Those scripts need to be run each time that docker starts / restarts, because the call to `docker events` exits when the docker daemon process stops. It would be nice to automate that.

I don't think such a facility exists in Docker Desktop for Mac at the moment?

If it does, please point me at it and close this!

I've written my own workaround: https://github.com/Mahoney/docker-lifecycle-listener - but I'd much rather use something out of the box.

**Comment by stephen-turner** [comment]
No, there's no such facility at the moment. So I'm going to move this ticket into docker/roadmap, which is the best place for feature requests.
